sdk/harambe/contrib/soup/impl.py

```diff
@@ -86,6 +86,21 @@ class SoupResponseWithStatus:
 
         return SoupResponseWithStatus()
 
+    async def post(self, url: str, data: dict[str, Any], **kwargs: Any) -> Any:
+        res = await self._session.post(
+            url, headers=self._extra_headers, data=data, **kwargs, impersonate="chrome"
+        )
+        if self._tracer:
+            self._tracer.log_request(res)
+
+        self._url = res.url
+        self._soup = BeautifulSoup(await res.text, "html.parser")
```

Comment:
depending on the content type in the response we may not want to change the _soup. eg if the post returns with application/json